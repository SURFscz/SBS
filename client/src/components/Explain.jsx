import React from "react";
import I18n from "i18n-js";
import "./Explain.scss";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";

export default class Explain extends React.PureComponent {

    componentWillReceiveProps(nextProps) {
        if (this.props.isVisible === false && nextProps.isVisible === true) {
            setTimeout(() => this.main.focus(), 50);
        }
    }

    render() {
        const {close, isVisible, subject, children} = this.props;
        return (
            <div className={`mod-explain ${isVisible ? "" : "hide"}`}
                 tabIndex="1" onBlur={close}
                 ref={ref => this.main = ref}>
                <section className="container">
                    <section className="header">
                        <p>{I18n.t("explain.title", {subject: subject})}</p>
                        <FontAwesomeIcon icon="window-close" onClick={close}/>
                    </section>
                    {children}
                </section>
            </div>
        );
    }
}
