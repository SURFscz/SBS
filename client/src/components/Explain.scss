@import "../stylesheets/vars.scss";

.mod-explain {

    position: fixed;
    top: 0;
    left: 0;
    box-shadow: 0 2px 4px 0 rgba(0, 0, 0, .1);
    background: white;
    width: 460px;
    z-index: 5;
    height: 100%;
    transition: left 325ms ease-in-out;

    &:focus {
        outline: none;
    }

    &.hide {
        transition: left 250ms ease-in-out;
        left: -470px;
    }

    .container {
        overflow-y: scroll;
        height: 99%;
    }

    p {
        margin-top: 5px;
    }

    span {
        &.emphasize {
            font-style: italic;
        }

        &.strong {
            font-weight: bold;
        }
    }

    .header {
        display: flex;
        background-color: $primary-green;
        color: white;
        font-size: larger;

        p {
            padding: 20px 15px;
        }

        svg {
            margin-left: auto;
            font-size: 32px;
            padding: 8px;
            cursor: pointer;
        }
    }

    .explanation, .example, .details {
        padding: 15px;
        font-size: 14px;
    }

}
