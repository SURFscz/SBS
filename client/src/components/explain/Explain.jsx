import React from "react";
import I18n from "../../locale/I18n";
import "./Explain.scss";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";

export default class Explain extends React.Component {

    innerClose = () => {
        const {close} = this.props;
        setTimeout(close, 200);
    }

    render() {
        const {close, isVisible, subject, children} = this.props;
        if (isVisible) {
            setTimeout(() => this.main && this.main.focus(), 150);
        }
        return (
            <div className={`mod-explain ${isVisible ? "" : "hide"}`}
                 tabIndex="1"
                 onBlur={this.innerClose}
                 onKeyDown={e => e.keyCode === 27 && this.innerClose()}
                 ref={ref => this.main = ref}>
                <section className="container">
                    <section className="header">
                        <p>{I18n.t("explain.title", {subject: subject})}</p>
                        <FontAwesomeIcon icon="times" onClick={close}/>
                    </section>
                    {children}
                </section>
            </div>
        );
    }
}
