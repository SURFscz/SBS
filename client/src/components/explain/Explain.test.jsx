import React from "react";
import {render} from "@testing-library/react";
import Explain from "./Explain";

jest.mock('@fortawesome/react-fontawesome', () => ({
    FontAwesomeIcon: () => <span data-testid="fa-icon"/>
}));

describe("Explain", () => {
    it("matches snapshot with default props", () => {
        const {asFragment} = render(<Explain/>);
        expect(asFragment()).toMatchSnapshot();
    });
});
