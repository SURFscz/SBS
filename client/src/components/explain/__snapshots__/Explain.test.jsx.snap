// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explain matches snapshot with default props 1`] = `
<DocumentFragment>
  <div
    class="mod-explain hide"
    tabindex="1"
  >
    <section
      class="container"
    >
      <section
        class="header"
      >
        <p>
          Explanation [missing "{{subject}}" value]
        </p>
        <span
          data-testid="fa-icon"
        />
      </section>
    </section>
  </div>
</DocumentFragment>
`;
